src/compiler/checker.ts

```diff
@@ -3677,11 +3678,25 @@ namespace ts {
                 const additionalContainers = mapDefined(container.declarations, fileSymbolIfFileSymbolExportEqualsContainer);
                 const reexportContainers = enclosingDeclaration && getAlternativeContainingModules(symbol, enclosingDeclaration);
                 const objectLiteralContainer = getVariableDeclarationOfObjectLiteral(container, meaning);
-                if (enclosingDeclaration && getAccessibleSymbolChain(container, enclosingDeclaration, SymbolFlags.Namespace, /*externalOnly*/ false)) {
+                if (enclosingDeclaration && container.flags & getQualifiedLeftMeaning(meaning) && getAccessibleSymbolChain(container, enclosingDeclaration, SymbolFlags.Namespace, /*externalOnly*/ false)) {
                     return append(concatenate(concatenate([container], additionalContainers), reexportContainers), objectLiteralContainer); // This order expresses a preference for the real container if it is in scope
                 }
+                // we potentially a symbol which is a member of the instance side of something - look for a variable in scope with the container's type
+                // which may be acting like a namespace
+                const firstVariableMatch = !(container.flags & getQualifiedLeftMeaning(meaning))

```

Comment:
This bit and the following modify our symbol container lookup to prefer, eg, `Symbol.iterator` to `SymbolConstructor.iterator` when doing printback for where the specified `SymbolFlags` say the first is allowable. Now, this isn't _specific_ to symbols or `Symbol` in any way - other dynamic names from namespace-ish things should work similarly, too! So `ArrayConstructor`, `Int8ArrayConsturctor`.... anything with the split namespace pattern the DOM uses a lot.